tests/caps_word/caps_word_autoshift/test_caps_word_autoshift.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 Google LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "keyboard_report_util.hpp"
+#include "keycode.h"
+#include "test_common.hpp"
+#include "test_fixture.hpp"
+#include "test_keymap_key.hpp"
+
+using ::testing::_;
+using ::testing::AnyNumber;
+using ::testing::AnyOf;
+using ::testing::InSequence;
+
+class CapsWord : public TestFixture {
+   public:
+    void SetUp() override {
+        caps_word_off();
+    }
+
+    // Convenience function to tap `key`.
+    void TapKey(KeymapKey key, int delay_ms = 0) {
+        key.press();
+        run_one_scan_loop();
+        if (delay_ms) {
+            idle_for(delay_ms);
+        }
+        key.release();
+        run_one_scan_loop();
+    }
```

Comment:
How about moving these helper methods into the base `TestFixture` class? I have seen it added in other unit-tests as well and future unit-tests would surely profit from them as well.

Reply:
Done. Added `tap_key(key, delay_ms)` and `tap_keys(key_a, key_b, ...)` methods to TestFixture.